monai/metrics/utils.py

```
@@ -291,13 +291,43 @@ def get_surface_distance(
-            dis = distance_transform_cdt(convert_to_numpy(~seg_gt), metric=distance_metric)
-        else:
-            raise ValueError(f"distance_metric {distance_metric} is not implemented.")
-    dis = convert_to_dst_type(dis, seg_pred, dtype=lib.float32)[0]
-    return dis[seg_pred]  # type: ignore
+            # Проверяем, находятся ли данные на CPU (numpy или torch на cpu)
+            is_cpu = isinstance(seg_pred, np.ndarray) or (not seg_pred.is_cuda)
+
+            if is_cpu:
+                from scipy.spatial import KDTree
+
+                # Переводим маски в numpy для работы с scipy
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `с` (CYRILLIC SMALL LETTER ES). Did you mean `c` (LATIN SMALL LETTER C)?

(RUF003)